ogusa/tests/test_SS.py

```diff
@@ -0,0 +1,163 @@
+        baseline, fsolve_flag, baseline_spending)
+
+    expected_dict = pickle.load(open(os.path.join(
+        CUR_PATH, 'test_io_data/SS_solver_outputs.pkl'), 'rb'))
+
+    for k, v in expected_dict.iteritems():
+        assert(np.allclose(test_dict[k], v))
+
+
+def test_inner_loop():
+    # Test SS.inner_loop function.  Provide inputs to function and
+    # ensure that output returned matches what it has been before.
+    input_tuple = pickle.load(open(os.path.join(
+        CUR_PATH, 'test_io_data/inner_loop_inputs.pkl'), 'rb'))
+    (outer_loop_vars, params, baseline, baseline_spending) = input_tuple
+    test_tuple = SS.inner_loop(
+         outer_loop_vars, params, baseline, baseline_spending)
+
+    expected_tuple = pickle.load(open(os.path.join(
+        CUR_PATH, 'test_io_data/inner_loop_outputs.pkl'), 'rb'))
+
+    for i, v in enumerate(expected_tuple):
+        assert(np.allclose(test_tuple[i], v))
+
+
+def test_euler_equation_solver():
+    # Test SS.inner_loop function.  Provide inputs to function and
+    # ensure that output returned matches what it has been before.
+    input_tuple = pickle.load(open(os.path.join(
+        CUR_PATH, 'test_io_data/euler_eqn_solver_inputs.pkl'), 'rb'))
+    (guesses, params) = input_tuple
+    test_list = SS.euler_equation_solver(guesses, params)
+
+    expected_list = pickle.load(open(os.path.join(
+        CUR_PATH, 'test_io_data/euler_eqn_solver_outputs.pkl'), 'rb'))
+
+    assert(np.allclose(np.array(test_list), np.array(expected_list)))
+
+
+def test_create_steady_state_parameters():
+    # Test that SS parameters creates same objects with same inputs.
+    input_dict = pickle.load(open(os.path.join(
+        CUR_PATH, 'test_io_data/create_params_inputs.pkl'), 'rb'))
+    test_tuple = SS.create_steady_state_parameters(**input_dict)
+
+    expected_tuple = pickle.load(open(os.path.join(
+        CUR_PATH, 'test_io_data/create_params_outputs.pkl'), 'rb'))
+
+    for i, v in enumerate(expected_tuple):
+        for i2, v2 in enumerate(v):
+            try:
+                assert(all(test_tuple[i][i2]==v2))
+            except ValueError:
+                assert((test_tuple[i][i2]==v2).all())
+            except TypeError:
+                assert(test_tuple[i][i2]==v2)
+
+
+@pytest.mark.parametrize('input_path,expected_path',
+                         [('run_SS_open_unbal_inputs.pkl',
+                           'run_SS_open_unbal_outputs.pkl'),
+                          ('run_SS_closed_balanced_inputs.pkl',
+                           'run_SS_closed_balanced_outputs.pkl')],
+                         ids=['Open, Unbalanced', 'Closed Balanced'])
+def test_run_SS(input_path, expected_path):
+    # Test SS.run_SS function.  Provide inputs to function and
+    # ensure that output returned matches what it has been before.
+    input_tuple = pickle.load(open(os.path.join(
+        CUR_PATH, 'test_io_data', input_path), 'rb'))
+    (income_tax_params, ss_params, iterative_params, chi_params,
```

Comment:
You may want to do:
```
with open(...) as f:
    input_tuple = pickle.load(f)
```

You get some nice behavior regarding error handling and closing the file connection once the data has been read.